File: tournaments/splitByValue/splitByValue.js
function splitByValue(k, elements) {
    const n = elements.length;
    const result = new Array(n);
    let index = 0;

    for (const item of elements) {
        if (k > item) {
            result[index++] = item;
        }
    }

    for (const item of elements) {
        if (k <= item) {
            result[index++] = item;
        }
    }

    return result;
}

File: tournaments/splitByValue/splitByValue.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub splitByValue {
    my ($k, $elements) = @_;

    my $n = @$elements;
    my @result = (0)x$n;
    my $index = 0;

    foreach my $item (@$elements) {
        if ($k > $item) {
            $result[$index++] = $item;
        }
    }

    foreach my $item (@$elements) {
        if ($k <= $item) {
            $result[$index++] = $item;
        }
    }

    return \@result;
}
